src/shared/credentials/defaultCredentialSelectionDataProvider.ts

```diff
@@ -59,62 +69,91 @@ export class DefaultCredentialSelectionDataProvider implements CredentialSelecti
+            ignoreFocusOut: true,
             shouldResume: this.shouldResume.bind(this),
+            buttons: [this.helpButton],
         })
+        return typeof result === 'string' ? result : undefined
     }
 
     public async inputAccessKey(
         input: MultiStepInputFlowController,
         state: Partial<CredentialSelectionState>
     ): Promise<string | undefined> {
-        return await input.showInputBox({
-            title: localize('AWS.title.createCredentialProfile', 'Create a new {0} credential profile', getIdeProperties().company),
+        const result = await input.showInputBox({
+            title: localize(
+                'AWS.title.createCredentialProfile',
+                'Create a new {0} credential profile',
+                getIdeProperties().company
+            ),
             step: 2,
             totalSteps: 3,
             value: '',
             prompt: localize('AWS.placeHolder.inputAccessKey', 'Input the {0} Access Key', getIdeProperties().company),
             validate: this.validateAccessKey.bind(this),
             ignoreFocusOut: true,
             shouldResume: this.shouldResume.bind(this),
+            buttons: [this.helpButton],
         })
+        return typeof result === 'string' ? result : undefined
     }
 
     public async inputSecretKey(
         input: MultiStepInputFlowController,
         state: Partial<CredentialSelectionState>
     ): Promise<string | undefined> {
-        return await input.showInputBox({
-            title: localize('AWS.title.createCredentialProfile', 'Create a new {0} credential profile', getIdeProperties().company),
+        const result = await input.showInputBox({
+            title: localize(
+                'AWS.title.createCredentialProfile',
+                'Create a new {0} credential profile',
+                getIdeProperties().company
+            ),
             step: 3,
             totalSteps: 3,
             value: '',
             prompt: localize('AWS.placeHolder.inputSecretKey', 'Input the {0} Secret Key', getIdeProperties().company),
             validate: this.validateSecretKey.bind(this),
             ignoreFocusOut: true,
             shouldResume: this.shouldResume.bind(this),
+            buttons: [this.helpButton],
         })
+        return typeof result === 'string' ? result : undefined
     }
 
-    public async validateNameIsUnique(name: string): Promise<string | undefined> {
+    public async validateProfileName(name: string): Promise<string | undefined> {
+        if (name === '') {
+            return localize('AWS.credentials.error.emptyProfileName', 'Profile name must not be empty')
+        }
+
         const duplicate = this.existingProfileNames.find(k => k === name)
 
         return duplicate ? 'Name not unique' : undefined
     }
 
     public async validateAccessKey(accessKey: string): Promise<string | undefined> {
         // TODO: is there a regex pattern we could use?
```

Comment:
Not your comment, but you can use the git-secrets regex (see the `--register-aws` flag: https://github.com/awslabs/git-secrets#operation-modes . Unsure if there's a secret key flag, but there definitely is for access.

Reply:
Interesting, good find. I am hesitant to use this however at least in the current blocking validation we do. Even within the docs for git-secrets it's called out that...

`While the patterns registered by this command should catch most instances of AWS credentials, these patterns are not guaranteed to catch them all.`

Unless we can get authoritative regexes from the IAM team we probably wouldn't want to do more than provide guidance ("Hey, it looks like your access key may be invalid. Might want to check on that.") vs. inadvertently blocking something that is actually correct (though even the aforementioned guidance could lead to confusion).

Reply:
Perhaps shorten this to `[A-Z0-9]{20}` and just cut the prefixes?

Reply:
Even this I'd be hesitant to add since as far as I can find this isn't a documented length constraint (the API states a [minimum of 16 and max of 128 characters](https://docs.aws.amazon.com/IAM/latest/APIReference/API_AccessKey.html)) and such a constraint risks being forward incompatible.